apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts

```diff
@@ -0,0 +1,333 @@
+import { html, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+export interface AdministratorQueryRunParams {
+  name: string;
+  sql: string;
+  params: string;
+  authn_user_id: string;
+  error: string | null;
+  result: AdministratorQueryResult | null;
+  formatted_date: string | null;
+}
+
+export interface AdministratorQueryResult {
+  rows: any[];
+  columns: string[];
+  rowCount: number;
+}
+
+export interface AdministratorQuery {
+  description: string;
+  resultFormats: any | null;
+  comment: any | null;
+  params: {
+    name: string;
+    description: string;
+    default: string | null;
+  }[];
+  sqlFilename: string;
+  link: string;
+}
+
+export function AdministratorQuery({
+  resLocals,
+  has_query_run,
+  query_run_id,
+  formatted_date,
+  params,
+  error,
+  result,
+  sqlFilename,
+  info,
+}: {
+  resLocals: Record<string, any>;
+  has_query_run: boolean;
+  query_run_id: string | null;
+  formatted_date: string | null;
+  params: AdministratorQueryRunParams | null;
+  error: string | null;
+  result: AdministratorQueryResult | null;
+  sqlFilename: string;
+  info;
+}) {
+  function renderHeader(columns, col) {
+    const row = {};
+    columns.forEach((c) => {
+      row[c] = true;
+    });
+
+    if (col === 'course_id' && 'course' in row) {
+      null;
```

Comment:
Should this be something like `return '';`?

Reply:
Similar concern in other lines with the same construct.

Reply:
Resolved in commit 227f6c3.